functions/Test-DbaRepLatency.ps1

```diff
@@ -0,0 +1,230 @@
+
+            # Step 1: Connect to the distributor
+            Write-Message -Level Verbose -Message "Connecting to Distributor"
+
+            try {
+                $distServer = Connect-SqlInstance -SqlInstance $DistributionServer -SqlCredential $SqlCredential -MinimumVersion 9
+            }
+            catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $DistributionServer -Continue
+            }
+
+            foreach($publication in $publicationNames) {
+
+                $pubMon = New-Object Microsoft.SqlServer.Replication.PublicationMonitor
+
+                $pubMon.Name = $publication.PublicationName
+                $pubMon.DistributionDBName = $distributionDatabase 
+                $pubMon.PublisherName = $publication.Server
+                $pubMon.PublicationDBName = $publication.Database
+
+                $pubMon.ConnectionContext = $distServer.ConnectionContext.SqlConnectionObject;
+
+
+                # Call the LoadProperties method to get the properties of the object. If this method returns false, either the publication monitor properties in Step 3 were defined incorrectly or the publication does not exist.
+                if(!$pubMon.LoadProperties()) {
+                    Stop-Function -Message "LoadProperties() failed. The publication does not exist." -Continue
+                }
+
+                $tokenList = $pubMon.EnumTracerTokens()
+
+                if(!$DisplayTokenHistory) {
+                    $tokenList = $tokenList[0]
+                }
+
+
+                foreach($token in $tokenList) {
+
+                    $tracerTokenId = $token.TracerTokenId
+
+                    $tokenInfo = $pubMon.EnumTracerTokenHistory($tracerTokenId)
+
+                    $timer = 0;
+                
+                    while($tokenInfo.Tables[0].Rows[0].subscriber_latency -eq [System.DBNull]::Value) {
+                        if($TimeToLive -and ($timer -gt $TimeToLive)) {
+                            break;
+                        }                        
+
+                        Start-Sleep -s 1
+                        $timer += 1
+                        $tokenInfo = $PubMon.EnumTracerTokenHistory($tracerTokenId)
+                    }
+                    
+
+                    foreach($info in $tokenInfo.Tables[0].Rows) {
+
+                        $totalLatency = if (($info.distributor_latency -eq [System.DBNull]::Value) -or ($info.subscriber_latency -eq [System.DBNull]::Value)) {
+                                                [System.DBNull]::Value
+                                            }
+                                            else {
+                                                 ($info.distributor_latency + $info.subscriber_latency)
+                                            }
+
+                        [PSCustomObject]@{
+                            TokenID      = $tracerTokenId
```

Comment:
Please don't forget our require fields - ComputerName, InstanceName and SqlInstance. I will update our other replication commands to follow this format too.

````powershell
                        ComputerName = $server.ComputerName
                        InstanceName = $server.InstanceName
                        SqlInstance  = $server.SqlInstance
````